addons/mrp/mrp.py

```diff
@@ -855,14 +855,12 @@ def test_production_done(self, cr, uid, ids):
                 res = False
         return res
 
-    def _get_subproduct_factor(self, cr, uid, production_id, move_id=None, context=None):
-        """ Compute the factor to compute the qty of procucts to produce for the given production_id. By default,
-            it's always equal to the quantity encoded in the production order or the production wizard, but if the
-            module mrp_subproduct is installed, then we must use the move_id to identify the product to produce
-            and its quantity.
-        :param production_id: ID of the mrp.order
+    def _get_subproduct_factor(self, cr, uid, move_id, context=None):
```

Comment:
if you create the `_calculate_produce_line_qty` method, there is no need to maintain an ugly hook method in mrp, just remove it!